src/notifications/lib/notification_model.py

```diff
@@ -7,6 +7,11 @@
 from datetime import datetime
 from pydantic import BaseModel, Field, HttpUrl, RootModel, model_validator
 
+class CustomBaseModel(BaseModel):
+    @model_validator(mode="before")
+    def replace_empty_lists_with_none(cls, values):
+        # Replace empty lists with None for all fields
+        return {k: (None if isinstance(v, list) and not v else v) for k, v in values.items()}
```

Comment:
This is probably the best way to do this, but unfortunately mode="before" can be any type of input. (See the paragraph below the code references in https://docs.pydantic.dev/latest/concepts/validators/#model-validators)

Can you add a isinstance(values, dict) check for this? I don't think we'll call it via Model.construct() way, but best to be safe. (Feel free to throw an exception if it's not a dict)